test/test_meta.py

```diff
@@ -539,6 +539,13 @@ def run_meta_crossref(
     try:
         rs = func(*args, **kwargs)
     except Exception as e:
+        if "foreach" in func.__name__:
```

Comment:
Why do we error if the original function is broken? Is it to save CI cost? It feels fine to just pass the meta reg if the original function is broken.

Specifically for foreach ops, lots of the sample inputs are more there to test parity in functionality including erroring when the nonforeach counterpart errors. It is not easy for every op to tease out all the possible error cases, so this is tested in one test case (test_parity). Thus, it results that the OpInfo for foreach ops will sometimes contain error inputs, which would fail in the meta test here.

This is not my favorite solution, but the alternatives don't seem better:
1. we could remove the error inputs from the foreach sample inputs --> this loses coverage, we still want to test that the erroring behavior is good
2. we could take on the bigger project of teasing out which inputs would fail vs not (this is nontrivial and will take time to refactor the infra.
3. we could leave these tests as skipped --> this also loses test coverage.

@awgu do you have thoughts? If landing the meta registrations is urgent, we can just add test skips before deciding as well.

Reply:
I realized we already set the precedent for just skipping the tests, opened https://github.com/pytorch/pytorch/issues/123663 to track